sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -133,17 +139,17 @@ public Mono<HttpResponse> send(HttpRequest request, Context context) {
                     return Mono.just(response);
                 }
             })
-            .onErrorMap(throwable -> {
+            .onErrorResume(throwable -> {
                 // The exception was an SSLException that was caused by a failure to connect to a proxy.
                 // Extract the inner ProxyConnectException and propagate that instead.
-                if (throwable instanceof SSLException && throwable.getCause() instanceof ProxyConnectException) {
-                    return throwable.getCause();
+                if (!proxyRetry && addProxyHandler && throwable instanceof ProxyConnectException
+                    || (throwable instanceof SSLException && throwable.getCause() instanceof ProxyConnectException)) {
```

Comment:
Does it improve the readability if we move the _ProxyConnect_ error check to its own function?

```java
if (isProxyConnectError(proxyRetry, throwable)) {
   return attemptAsync(..);
}
```

Reply:
Will add this in the next commit